classes/class-query.php

```diff
@@ -246,12 +246,29 @@ public function query( $args ) {
 		 */
 		$query = apply_filters( 'wp_stream_db_query', $query, $args );
 
-		$result = array();
+		// Build result count query.
+		$count_query = "SELECT COUNT(*) as found
+		FROM $wpdb->stream
+		{$join}
+		WHERE 1=1 {$where}";
+
+		/**
+		 * Filter allows the result count query to be modified before execution.
+		 *
+		 * @param string $query
+		 * @param array  $args
+		 *
+		 * @return string
+		 */
+		$count_query = apply_filters( 'wp_stream_db_count_query', $count_query, $args );
+
 		/**
 		 * QUERY THE DATABASE FOR RESULTS
 		 */
-		$result['items'] = $wpdb->get_results( $query ); // @codingStandardsIgnoreLine $query already prepared
-		$result['count'] = $result['items'] ? absint( $wpdb->get_var( 'SELECT FOUND_ROWS()' ) ) : 0;
+		$result = array(
+			'items' => $wpdb->get_results( $query ), // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+			'count' => absint( $wpdb->get_var( $count_query ) ), // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
```

Comment:
@ivankruchkoff @kasparsd @derekherman I added explicit ignore comments here. I think don't anymore sanitation needs to be done because all dynamic parts of the query are run thru `$wpdb->prepare()` earlier in the function.

Reply:
Yeah, all the dynamic parts seem to be fine. However, we're now passing those queries through filters as the very last thing. I guess we need to trust that whoever is using that filter will do their job to sanitize the adjusted parts.